### ISSUE TITLE
InputEvent's device id is always 0

### ISSUE THREAD

**Issue by flurick** [godotengine/godot#23439]
**Godot version:** 3.0.5stable & 3.1alpha (13eaded)

**OS/device including version:** Linux 4.18.16

**Issue description:** The event’s device ID is always set to 0, was hoping to detect what keyboard or mouse was creating the InputEvent from in "func _input(event):..."

**Steps to reproduce:** Add a script to a node and and have it access the InputEvents "device" property.

**Minimal reproduction project:**[InputEvents device id is always 0.zip](https://github.com/godotengine/godot/files/2539757/InputEvents.device.id.is.always.0.zip)

**Comment by Naryosha** [comment]
Can confirm on Manjaro x86_64. Also tried with 2 keyboards, still 0.

Looking from the source, it seems like `device` is not altered in most platforms.

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in [Godot 3.2.2](https://godotengine.org/article/maintenance-release-godot-3-2-2) or any later release?

**Comment by KoBeWi** [comment]
Closing due to lack of update. Please comment if you can still reproduce this issue in the latest stable version of Godot.

**Comment by beroso** [comment]
This bug persists in the current version (v3.3.1.stable)

**Comment by KoBeWi** [comment]
@beroso What OS are you on?

**Comment by beroso** [comment]
> @beroso What OS are you on?

MacOS Catalina 10.15.7

**Comment by beroso** [comment]
> This bug persists in the current version (v3.3.1.stable). Tested with keyboard and a joystick, the event device is zero for both

Maybe this behavior is intended by design, but for me it was not obvious. I didn't find documentation about that.
Obs: I did not try with 2 keyboards or two joysticks, but I think it works as intended.

**Comment by KoBeWi** [comment]
Yeah I think this method is mostly useful for joysticks. Maybe it just needs documentation change.

**Comment by madmiraal** [comment]
A device id is relevant anytime there are multiple input devices of the same type. This doesn't just apply to joysticks. See https://github.com/godotengine/godot/issues/53033#issuecomment-1003364565 and https://github.com/godotengine/godot-proposals/issues/426 for a reason why being able to identify between different pointer (Mouse Input) devices is important.
